stubs/redis/redis/typing.pyi

```diff
@@ -0,0 +1,33 @@
+from collections.abc import Iterable
+from datetime import datetime, timedelta
+from typing import Protocol, TypeVar
+from typing_extensions import TypeAlias
+
+from redis.asyncio.connection import ConnectionPool as AsyncConnectionPool
+from redis.connection import ConnectionPool
+
+EncodedT: TypeAlias = bytes | memoryview
+DecodedT: TypeAlias = str | int | float
+EncodableT: TypeAlias = EncodedT | DecodedT
+AbsExpiryT: TypeAlias = int | datetime
+ExpiryT: TypeAlias = float | timedelta
+ZScoreBoundT: TypeAlias = float | str
+BitfieldOffsetT: TypeAlias = int | str
+_StringLikeT: TypeAlias = bytes | str | memoryview
+KeyT: TypeAlias = _StringLikeT
+PatternT: TypeAlias = _StringLikeT
+FieldT: TypeAlias = EncodableT
+KeysT: TypeAlias = KeyT | Iterable[KeyT]
+ChannelT: TypeAlias = _StringLikeT
+GroupT: TypeAlias = _StringLikeT
+ConsumerT: TypeAlias = _StringLikeT
+StreamIdT: TypeAlias = int | _StringLikeT
+ScriptTextT: TypeAlias = _StringLikeT
+TimeoutSecT: TypeAlias = int | float | _StringLikeT
+AnyKeyT = TypeVar("AnyKeyT", bytes, str, memoryview)  # noqa: Y001
```

Comment:
I assume these `TypeVar`s have public names because they have public names in the source code — could we add a comment to that effect?